tmt/steps/provision/mrack.py

```diff
@@ -197,6 +206,34 @@ class MrackHWNotGroup(MrackHWGroup):
     name: str = 'not'
 
 
+def _get_custom_config() -> dict[str, Any]:
+    if TMT_MRACK_HW_FILTERS_FILE.is_absolute():
+        config_content = TMT_MRACK_HW_FILTERS_FILE.read_text()
+    else:
+        config_content = (tmt.utils.Config().path / TMT_MRACK_HW_FILTERS_FILE).read_text()
+    return tmt.utils.yaml_to_dict(config_content)
+
+
+def _translate_constraint_by_config(
+        constraint: tmt.hardware.Constraint[Any],
+        translations: list[Any]
+        ) -> dict[str, Any]:
+    for translation in translations:
```

Comment:
It would be extremely useful to include an example config file. It should be small, well-documented, showing an example or two - commented out if needed. Right now, all we have is guessing based on the unit test below.

Another point I'm not sure about: what should the file contain? XML, or the simplified node-like mappings we have above? Or something else? E.g. we have something like this in Artemis:

```yaml
  compatible:
    distro:
      translations:
        - operator: contains 
          value: rhel-6
          element: '<compatible_with_distro arch="<< ENVIRONMENT.hw.arch >>" osmajor="RedHatEnterpriseLinux6"/>'
```

I.e. maps `compatible.distro` constraint to an actual XML. It's needlessly complicated, we could have made it simpler, but there needs to be a selector and the pattern, and I'm thinking about generating the actual XML, with Jinja because the use of templates proved to be very valuable - without templates, it'd be very hard to inject guest architecture into the element above. Simplified, it might be something like this (with `GUEST` representing the tmt guest for which we are generating the filter):

```yaml
compatible.distro: |
  {% if GUEST.image | regex_match('(?i)rhel-6') %}
    <compatible_with_distro arch="{{ GUEST.arch }}" osmajor="RedHatEnterpriseLinux6"/>'
  {% endif %}
```

We could get rid of `operator` and `translation` and leave it to Jinja. `operator` as a concept did not prove to be very flexible in Artemis implementation - this sucks:

```yaml
    is_virtualized:
      translations:
        - operator: '=='
          value: false
          element: '<system><hypervisor op="==" value="" /></system>'

        - operator: '=='
          value: true
          element: '<system><hypervisor op="!=" value="" /></system>'
```

What I don't know is how to take the output of such a template, and connect it with the tree or dataclasses we build. Dictionaries would be easier, and you took that route, but those don't make the use of templates easier:

```yaml
compatible.distro:
  compatible_with_distro:
    arch: "{{ GUEST.arch }}"
    osmajor: {% if GUEST.image | regex_match('(?i)rhel-6') %}RedHatEnterpriseLinux6{% endif %}

# Or, the value is nothing but a template string, we then `yaml_to_dict()` it once more:
compatible.distro: |
  compatible_with_distro:
    arch: "{{ GUEST.arch }}"
    osmajor: {% if GUEST.image | regex_match('(?i)rhel-6') %}RedHatEnterpriseLinux6{% endif %}
```

In other words, how can we make the filter parametrized so we wouldn't have to enumerate every single distro and arch combination for `compatible.distro`? :)